doc/man7/provider.pod

```diff
@@ -42,7 +42,7 @@ If it's built-in, the initialization function may have any name.
 
 The initialization function must have the following signature:
 
- int NAME(const OSSL_PROVIDER *provider,
+ int NAME(const OSSL_CORE_PROVIDER *provider,
```

Comment:
I wonder if it shouldn't be called something like `OSSL_CORE_HANDLE`...  there is no "core provider", and the only use this type has anyway is as a handle to the core for some of the upcalls...

Reply:
I've done this rename